nixos/modules/services/backup/syncoid.nix

```diff
@@ -0,0 +1,151 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.syncoid;
+in {
+
+    # Interface
+
+    options.services.syncoid = {
+      enable = mkEnableOption "Syncoid ZFS synchronization service";
+
+      interval = mkOption {
+        type = types.str;
+        default = "hourly";
+        example = "*-*-* *:15:00";
+        description = ''
+          Run syncoid at this interval. The default is to run hourly.
+
+          The format is described in
+          <citerefentry><refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum></citerefentry>.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "root";
+        example = "backup";
+        description = ''
+          The user for the service. Sudo or ZFS privilege delegation must be
+          configured to use a user other than root.
+        '';
+      };
+
+      sshKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          SSH private key file to use to login to the remote system. Can be
```

Comment:
Since this should never end up in the nix store, I suggest doing this:
```nix
mkOption {
  apply = mapNullable toString;
  # ...
}
```

With this, when users pass `/foo/some/key`, it will get translated to `"/foo/some/key"`, which won't copy the key into the store, and requires users to provide it at runtime.